### PR TITLE
XRT-1221: fix for infinite loop of RPM decode

### PR DESCRIPTION
**Pull request by FraserG01**

### CODE REVIEW

File: src/bacnet/bacapp.c [hunk 1]
```diff
@@ -1721,7 +1691,6 @@ static int bacapp_decode_context_data_complex(
                     max_apdu_len - apdu_len, inner_tag_number, &tmpvalue, prop);
                 if (len >= 0) {
                     apdu_len += len;
-                    value = bacapp_allocate_new_value(&allocate_data, value, &tmpvalue);
```
Comment (edward-scott) [inline src/bacnet/bacapp.c:1724]: Probably should be renamed to bacapp_decode_context_data_complex_len or similar and the value param removed. Can be done later though.

**Review by edward-scott — APPROVED** [review]
See comment. LGTM.

File: src/bacnet/bacapp.c (hunk 1 above)
Comment (FraserG01) [reply src/bacnet/bacapp.c:1724]: Sure will add a note to do that

**Comment by SteveOss** [comment]
@FraserG01 Why was my comment not addressed ?

**Comment by FraserG01** [comment]
> @FraserG01 Why was my comment not addressed ?

@SteveOss What comment sorry? I only see Eds?

**Comment by SteveOss** [comment]
Seems to be in a pending state ? Comment was for src/bacnet/basic/service/h_rpm_a.c line 117:

Would be nicer to use a #define with a meaningful name rather than just tag 4 (same for all other tags). If this exists should use, if not would made sense to add as part of another "clean up" ticket.

**Comment by FraserG01** [comment]
> Seems to be in a pending state ? Comment was for src/bacnet/basic/service/h_rpm_a.c line 117:
> 
> Would be nicer to use a #define with a meaningful name rather than just tag 4 (same for all other tags). If this exists should use, if not would made sense to add as part of another "clean up" ticket.

Okay sure will add a ticket for both you and Ed's points
I dont see the comment at all

**Pull request merged**